providers/salesforce/register.go

```diff
@@ -0,0 +1,231 @@
+package salesforce
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/go-playground/validator"
+)
+
+var errInvalidRequestType = errors.New("invalid request type")
+
+type RegistrationParams struct {
+	// UniqueRef is a unique reference for the registration.
+	// It is used to create unique names for the Salesforce objects.
+	UniqueRef string `json:"uniqueRef" validate:"required"`
+	Label     string `json:"label"     validate:"required"`
+	AwsArn    string `json:"awsArn"    validate:"required"`
+}
+
+type ResultData struct {
+	EventChannel     *EventChannel
+	NamedCredential  *NamedCredential
+	EventRelayConfig *EventRelayConfig
+}
+
+func (c *Connector) RollbackRegister(ctx context.Context, res *ResultData) error {
+	if res.EventRelayConfig != nil {
+		_, err := c.DeleteEventRelayConfig(ctx, res.EventRelayConfig.Id)
+		if err != nil {
+			return fmt.Errorf("failed to delete event relay config: %w", err)
+		}
+	}
+
+	if res.NamedCredential != nil {
+		_, err := c.DeleteNamedCredential(ctx, res.NamedCredential.Id)
+		if err != nil {
+			slog.Error("failed to delete named credential", "error", err)
```

Comment:
Rather than using slog directly it's better to use `logging.Logger(ctx).Error(...)`

I won't type it out for each instance but this feedback applies to any direct uses of `slog.Whatever`